File: T7/Ejercicios Clase/Ejercicio 2 T6/src/model/Asalariado.java
package model;

public class Asalariado extends Trabajador{

    private double sueldo;
    private int nPagas;
    private boolean contratado;


    public Asalariado(){

    }

    public Asalariado(String nombre, String apellido, String dni, double sueldo, int nPagas, boolean contratado) {
        super(nombre, apellido, dni);
        this.sueldo = sueldo;
        this.nPagas = nPagas;
        this.contratado = contratado;
    }


    @Override
    public void mostrarDatos() {
        super.mostrarDatos();
        System.out.println(" Sueldo:" + sueldo + " Numero de pagas: " + nPagas + " Contratado: " + contratado);
    }

    public double getSueldo() {
        return sueldo;
    }

    public void setSueldo(double sueldo) {
        this.sueldo = sueldo;
    }

    public int getnPagas() {
        return nPagas;
    }

    public void setnPagas(int nPagas) {
        this.nPagas = nPagas;
    }

    public boolean isContratado() {
        return contratado;
    }

    public void setContratado(boolean contratado) {
        this.contratado = contratado;
    }
}

File: Curso 2/PMPD/TrabajadoresHerencia/src/model/Asalariado.kt
package model

class Asalariado(
    nombre: String,
    apellido: String,
    dni: String,
    salario: Double,
    var nPagas: Int,
    var irpf: Double
) : Trabajador(nombre, apellido, dni, salario){

    override fun calcularSalarioNeto(): Double {
        // salario total - lo que te quitan de IRPF
        var salarioNeto: Double = salario -(salario*irpf)

        return salarioNeto
    }

    // un asalariado puede pedir un aumento de sueldo
        // se genera un aleatorio entre 1-10
        // en caso de generar un numero <5 NO se le sube el salario
        // en caso de generar un numero >=5 SI se le sube un 10% el salario

    fun pedirAumento(){
        val random : Int = (1..10).random()

        if (random < 5) {
            println("Lo siento, no te subimos el sueldo")
        } else {
            var salarioAntiguo: Double = salario
            salario += (salario*0.10) // salario *= 1.1 es lo mismo
            println("Te subimos el sueldo un de $salarioAntiguo a $salario")
        }
    }

    override fun mostrarDatos() {
        super.mostrarDatos()
        println("nPagas: $nPagas")
        println("IRPF: $irpf")
    }

}